Added documentation of indexes for case-insensitive databases (#4209)
wso2/docs-is: en/identity-server/next/docs/includes/db-config-table.md

!!! info 
    For more information on other parameters that can be defined in
    the `<IS_HOME>/repository/conf/deployment.toml` file, see [Tomcat JDBC Connection Pool](http://tomcat.apache.org/tomcat-9.0-doc/jdbc-pool.html#Tomcat_JDBC_Enhanced_Attributes).

??? note "Support for case-sensitive usernames"

>     WSO2 Identity Server supports case-insensitive usernames by default. Hence, if you need to have case-sensitive usernames, you need to configure the following properties.

Reviewer: ```suggestion
    Usernames in {{product_name}} are case-insensitive by default. If you wish to enable case-sensitive usernames, configure the following properties.
```

>     For the primary user store, add the following configurations to the `<IS-HOME>/repository/conf/deployment.toml` file.

Reviewer: structure this similar to the notes above.

Author: https://github.com/wso2/docs-is/pull/4209/commits/d0b97b1a597730a4d099c4dc1565e31452c0f03f

    ``` toml
    [user_store.properties]
    CaseInsensitiveUsername = false
    UseCaseSensitiveUsernameForCacheKeys = false
    ```  
    
    For secondary user stores, add the following configurations to the `<userstore>.xml` file in the `<IS_HOME>/repository/deployment/server/userstores` folder.

    ``` xml
    <Property name="CaseInsensitiveUsername">false</Property>
    <Property name="UseCaseSensitiveUsernameForCacheKeys">false</Property>
    ```

>     The database indexes have been created using lower functions to support case-insensitive usernames for the related tables. Therefore, remove the lower functions from the related index creation queries in the scripts at `<IS_HOME>/dbscripts/`.

Reviewer: ```suggestion
    The database indexes are created using LOWER() functions to support case-insensitive usernames for the related tables. Therefore, remove the LOWER() functions from the related index creation queries in the scripts at `<IS_HOME>/dbscripts/`.
```

!!! info 
    For more information on other parameters that can be defined in
    the `<IS_HOME>/repository/conf/deployment.toml` file, see [Tomcat JDBC Connection Pool](http://tomcat.apache.org/tomcat-9.0-doc/jdbc-pool.html#Tomcat_JDBC_Enhanced_Attributes).

??? note "Support for Case Sensitive Usernames"

>     If you are going to have case-sensitive usernames, 

Reviewer 2: ```suggestion
    WSO2 Identity Server supports case-insensitive usernames by default. Hence, if you need to have case-sensitive usernames, you need to configure the following properties.
```

>     you need to add the following properties to the deployment.toml file.

Reviewer 2: Remove

Author: resolved with https://github.com/wso2/docs-is/pull/4209/commits/24ec8d148e9a0a8fa9efc843b02d9856a987e914

>     For Primary Userstores

Reviewer 2: ```suggestion
    For the primary user store, add the following configurations to the `<IS-HOME>/repository/conf/deployment.toml` file.
```

    ``` toml
    [user_store.properties]
    CaseInsensitiveUsername = false
    UseCaseSensitiveUsernameForCacheKeys = false
    ```  
    

>     For Secondary Userstores

Reviewer 2: ```suggestion
    For secondary user stores, add the following configurations to the `<userstore>.xml` file in the `<IS_HOME>/repository/deployment/server/userstores` folder.
```

    ``` xml
    <Property name="CaseInsensitiveUsername">false</Property>
    <Property name="UseCaseSensitiveUsernameForCacheKeys">false</Property>
    ```

>     Since by default IS supports case insensitive usernames, from the database level we created lower indexes for the related tables. 

Reviewer 2: ```suggestion
    The database indexes have been created using lower functions to support case-insensitive usernames for the related tables. Therefore, remove the lower functions from the related index creation queries in the scripts at `<IS_HOME>/dbscripts/`.
```

    Therefore remove those lower function when creating the indexes from the relevant db scripts.
    

>     For example, in the `database script` files in the `dbscripts` directory, remove the `lower` function from the queries.

Reviewer 2: Remove

Author: resolved with https://github.com/wso2/docs-is/pull/4209/commits/24ec8d148e9a0a8fa9efc843b02d9856a987e914
